buildSrc/src/main/groovy/org/mockito/workflow/gradle/internal/ReleaseWorkflowPlugin.java

```diff
@@ -0,0 +1,35 @@
+package org.mockito.workflow.gradle.internal;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.execution.TaskExecutionListener;
+import org.gradle.api.tasks.TaskState;
+import org.mockito.release.exec.Exec;
+import org.mockito.release.git.Git;
+import org.mockito.release.git.GitTool;
+
+public class ReleaseWorkflowPlugin implements Plugin<Project> {
+    public void apply(final Project project) {
+        //main release task
+        project.getTasks().create("release");
+        final ReleaseWorkflowExtension ext = project.getExtensions().create("releaseWorkflow", ReleaseWorkflowExtension.class, project);
+
+        //setup listener, so that the rollbacks are only executed if one of the main tasks fail
+        project.getGradle().addListener(new TaskExecutionListener() {
+            public void beforeExecute(Task task) {}
+            public void afterExecute(Task task, TaskState taskState) {
+                //when one of the main step tasks fails, enable all rollback tasks
+                if (taskState.getFailure() != null && ext.steps.contains(task)) {
+                    for (Task rollback : ext.rollbacks) {
+                        rollback.setEnabled(true);
+                    }
+                }
```

Comment:
I think this code could be a method in `ReleaseWorkflowExtension`. It doesn't look like OO at this moment.

```java
ext.rollbackIfNeeded(task, taskState);
```


```java
class ReleaseWorkflowExtension {
  // ...

  public void rollbackIfNeeded(Task task, TaskState taskState) {
    if (taskState.getFailure() != null && steps.contains(task)) {
      for (Task rollback : rollbacks) {
        rollback.setEnabled(true);
      }
    }
  }
}
```